Use AVAudioSession.requestRecordPermission before starting to recognize

nacho4d opened the pull request:
This solves #34 

Code is slightly different from https://github.com/mihui/speech-ios-sdk/commit/a5bd6aedf5af5466977d0b0e0f07b15f02ebb324 because it does not cache `isPermissionGranted`. It is not cached because the user could change permissions between recording sessions and then this flag would be invalid.

`[[AVAudioSession sharedInstance] requestRecordPermission:...` does not *really* request access when it has been granted already so it is ok to call it always.

I am not handling iOS6 since `IPHONEOS_DEPLOYMENT_TARGET` is 8.0 now. @mihui if you need ios6 I guess is doable, although it became 8 a bit ago (See https://github.com/watson-developer-cloud/speech-ios-sdk/issues/28)

mihui commented:
I see @nacho4d , it is OK to me.

Many of our internal projects are still supporting iOS 6, also some of my clients from Japan, that's the reason why I'm looking into compatibility issues of this SDK, though personally I prefer to upgrade to iOS 8 and above.

nacho4d commented:
@mihui I see. I have added ios6 compatibility (fixed code I changed some weeks ago) .
Can we close issue #28 
Resolution would be:

Static library target "watsonsdk" is now iOS6 and above again :p
Dynamic framework target "WatsonSDK" is iOS8 and above.

nacho4d commented:
@mihui I am merging this. I understand iOS6 is still important to you. I will add some scripts to make again the static framework we used to have that would be in another PR. Sorry for all the trouble caused.
